.github/workflows/auto-rel-merge.yml

```diff
@@ -0,0 +1,83 @@
+name: Auto Merge Changes from maint-* Branches
+
+on:
+  push:
+    branches:
+      - 'maint-*'  # Trigger when any maint-* branch is updated
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
```

Comment:
just to point out that this is not an oversight: actions/checkout will use the branch that triggered the event, so maint-6.6 in this case, which is what we want, cf https://github.com/actions/checkout?tab=readme-ov-file#usage

not sure if I'd prefer to specify the ref here explicitly, but as far as I am aware this is functionally equivalent